bundles/org.openhab.binding.freeathomesystem/README.md

```diff
@@ -0,0 +1,81 @@
+# ABB/Busch-free@home Smart Home binding
+
+ openHAB ABB/Busch-free@home binding based on the offical free@home local api
+
+![alt text](https://github.com/jannodeluxe/jannnnoooo/blob/main/free_at_home_logo_1.jpg)
+![alt text](https://github.com/jannodeluxe/jannnnoooo/blob/main/abb_freeathome_2_0.png)
+# Description
+
+This openHAB binding allows you to connect your free@home Smart Home system from ABB / Busch-Jaeger to openHAB and to control and observe most of the components.
+It requires a System Access Point with version 2.6.1 or higher.
+
+# Supported Devices
+
+- ABB / Busch-Jaeger System Access Point 2.0
+- free@home Switch Actuator Sensor 1/1, 2/1, 2/2 (wired and wireless)
+- free@home Dimming Actuator Sensor 1/1, 2/1 (wired and wireless)
+- free@home Blind Actuator Sensor 1/1, 2/1 (wired and wireless)
+- free@home Movement Detector Actuator
+- free@home Radiator Thermostat
+- free@home Switch Actuator 4-channel
+- free@home Switch Actuator 4-channel
+- free@home Dimming Actuator 4- and 6-channel
+- IP-touch panel (function: door opener, door ring sensor)
+- Hue devices (untested)
+
+
+# Tested SysAP Versions
+
+|Version|Supported|
+|---|---|
+|2.6.1+|yes|
+
+
+# Setup / Installation
+
+## Prerequisites
+
+To make use of this Binding first the local free@home API has to be activated. The API is disabled by default!
+
+1. Open the free@home next app
+2. Browse to "Settings -> free@home settings -> local API and activate the checkbox
+
+## Discovery
+
+The free@home bridge shall be added manually. Once it is added as a Thing with correct credentials, the scan of free@home devices will be possible.
+The devices with multiple channels are devided into multiple devices with a single channel. The label of these devices is built-up with the same deivce name and free@home device ID but the channel number is mentioned in the Thing label as well for better identification.
```

Comment:
What's the reason for splitting these into multiple Things?

Reply:
It was my first design idea to reflect identically the free@home devices in openHAB Things. But the free@home API is not reporting the device type (e.g. sensor actuator with 2 sensor and 2 actuator channels), the API is streaming out only how many and which channel types are available in the device. Additionally the device channels can have different settings on the free@home dashboard.
Due to these issues, the detection code to identify the device type based on its channels and configuration was a very complex and hard to maintainable code. Therefore I made this design decision to split the device channels into multiple Things. This makes the maintenance and the handler code simpler.

Reply:
Maybe doing stuff similarly to the Homematic binding would be an idea? It builds thing types dynamically based on the information about the channels gathered from the HM RPC API, so from your description it's about the same thing as in your case.

Reply:
I check the solution in the Homematic binding. Good idea. I would implement in a later version as an option for the users.

However in this implementation a general free@home device as thing must be available, because the free@home API is not reporting which device do I have, reporting only the device name and the type and number of channels
Therefore during the discovery on the openHAB dashboard the device can be displayed only with its name, but as a general device. (because it is not known)

After adding the device the user could open its channels and can see which device is hiding in this general device, (however the name can say something about the device type, which the user can configure in the free@home system dashboard)

Due to the general device the user is not able to sort the different kind of devices (Themostat, Switch-Actuator etc) but will see only a long list, which can be sorted based on name if the user gave correct names to the free@home system dashboard
With many devices 

It is a good idea to do like the Homematic binding, but I would do it only as an option. If it is done as an option the binding code shall not be more complicated because behind the general device the same program code can be used as used for the spited devices.

@maniac103 
What do you think is it possible to implement in the next version as an option?

Reply:
I honestly do not yet understand the problem. It seems the API for each devices gives you a JSON object with a (unique, I guess?) device ID and a description of channels belonging to that device ID. What is the problem with generating one thing per device ID? The thing type would need to be generated dynamically, like for HM, and the newly created thing would be assigned precisely that thing type. One could generate an artificial thing type ID based on the channels in it and thus reuse it for multiple devices with the same channel set.

What am I missing here? IMHO the translation from one API device to multiple things is somewhat weird, but at least very uncommon.

Reply:
Changing your code to the Homematic approach in the future, would break compatibility. As this binding is supposed to become part of the official distribution, it should stick to the defintion what a Thing is: Things are "physical entities". See https://www.openhab.org/docs/configuration/things.html

So, each physical sensor should be one Thing. If the device type is not provided by the API, it could be a generic Thing with dynamic Channels.

Reply:
I re-written the binding to create the things dynamically based on the detected free@home device properties. The channels are created also dynamically based for each thing.

I will make some last tests and will set the refactor some files for better readability, and will set the pull request again on

Thank you that you challenged me to make a better code